public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx

```diff
@@ -54,6 +82,39 @@ import {
 } from './reducer';
 import { useAlertQueryRunner } from './useAlertQueryRunner';
 
+export function areQueriesTransformableToSimpleCondition(
+  dataQueries: Array<AlertQuery<AlertDataQuery | ExpressionQuery>>,
+  expressionQueries: ExpressionQuery[]
+) {
+  if (dataQueries.length !== 1) {
+    return false;
+  }
+
+  if (expressionQueries.length !== 2) {
+    return false;
+  }
+
+  const query = dataQueries[0];
+
+  if (query.refId !== SIMPLE_CONDITION_QUERY_ID) {
+    return false;
+  }
+
+  const reduceExpression = expressionQueries.find(
+    (query) => query.type === ExpressionQueryType.reduce && query.refId === SIMPLE_CONFITION_REDUCER_ID
+  );
+  const reduceOk =
```

Comment:
I feel like there's something missing in this conditional; do we not need to assert if the reduce expression is at index `0` and the threshold expression is at index `1`?

Reply:
good catch!